Working

Hamsterrhino opened the pull request.

Hamsterrhino committed:
Starting new project

Hamsterrhino committed:
1 week probably done

drewxa commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,71 @@
+			std::string arg1 = argv[3];
+			std::string password;
+			std::cin >> password;
+			std::string sqstring = "INSERT INTO Passwords VALUES('" + arg + "','" + arg1 + "','" + password + "');";
```
возможна SQL инъекция

drewxa commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,71 @@
+}
+
+int main(int argc, char** argv) {
+	sqlite3* db = 0; // ����� ������� ���������� � ��
```
я бы какую-то библиотеку использовал бы для SqlLite

drewxa commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,71 @@
+	const char* data = "Callback function called";
+	char* zErrMsg = 0;
+	const char* sql;
+	// ��������� ����������
```
кодировку поправить

adkuz replied:
Или писать только на английском.

adkuz commented:
Необходимо убрать миллион бинарей, которые не являются вашим кодом.
Это метафайлы, которые создает VS? Я не знаю, что это, и знать не хочу. В системе контроля версий этого быть не должно.

adkuz commented:
Нэйминг коммитов и ПРов обсудим лично.
Оставляю этот коммент, чтобы самому не забыть.

adkuz reviewed (changes requested):
Очень тяжело читать: большая часть файлов - мусор, код даже трудно найти.
В самом коде нет комментариев, без чего сложно ориентироваться даже в таких нетрудных для понимания участках.

adkuz commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,71 @@
+
+				sqlite3_free(err);
+			}
+			std::string arg = argv[2];
+			std::string arg1 = argv[3];
```
Переименовать переменные.

adkuz commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,71 @@
+			std::cin >> password;
+			std::string sqstring = "INSERT INTO Passwords VALUES('" + arg + "','" + arg1 + "','" + password + "');";
+
+			sql = sqstring.c_str();
```
Смысл переменной?
Это запрос?

adkuz commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,71 @@
+#include <iostream>
+#include <string>
+
+int callback(void* NotUsed, int argc, char** argv, char** azColName) {
+	for (size_t i = 1; i < argc; i++) {
+		std::cout << azColName[i] << ": " << argv[i] << std::endl;
+	}
+	std::cout << std::endl;
+	return 0;
+}
```
Что эта функция делает? По сигнатуре понять почти нереально.

adkuz commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,71 @@
+		} break;
+		case 'l': {
+			std::string arg = argv[2];
+			std::string sqstring = "SELECT * from Passwords WHERE SOURCE = '" + arg +"';";
```
Та же уязвимость для инъекций.

adkuz reviewed (changes requested):
Мусор надо удалить и почистить решение в принципе.

Hamsterrhino committed:
Deleted 4 files,Started to work with querries

Hamsterrhino committed:
Week 2, finished

Got rid of Trash
Probably injection-proof
AES-128 added
Keyfile.txt is located in Debug

Hamsterrhino committed:
Deleted exe

Hamsterrhino committed:
Added encryption for password

Added encryption for passwords, for some reason decryption doesn't work yet.
Converted keyfile to txt file, so anyone can open it.

Hamsterrhino committed:
Same as before, forgot to update cpp file

Hamsterrhino committed:
Encryption and Decryption bodies are now in seperate hpp file

Hamsterrhino committed:
Put Encryptioni and Decryption in AES

Tried to fix the issue of failed decryption...
Failed.

Hamsterrhino committed:
Fixed Encryption and Decrpytion

Changed algorithm (now works properly), got rid of some unnecessary files.

drewxa commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,111 @@
+        std::ifstream keyfile("aes-master\\keyfile", std::ios::binary);
+        std::string keystring;
+        std::getline(keyfile, keystring);
+        unsigned char *key = new unsigned char[strlen(keystring.c_str()) + 1];
```
std::vector

Hamsterrhino replied:
Вопрос, а надо ли переводить char в вектор? Тогда придёться хранить в векторе по одному char'у, да и алгоритм надо будут переписать чутка, не говоря уже про непредвиденные ошибки.

drewxa commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,111 @@
+  char *zErrMsg = 0;
+  const char *sql;
+  // openning DB
+  if (sqlite3_open("my_cosy_database.db", &db))
```
Серьезно? sqlite3_open возвращает true если ошибка произошла?

Hamsterrhino replied:
sqlite3_open сам по себе, открывает базу данных. Если ошибки не нашлось, то он открывает её и идёт дальше. То есть
error=false
Если ошибка есть, то 
error=true
и выполняет алгоритм if.

drewxa commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,111 @@
+  // SQL execute
+  else {
+    if (argc > 1) {
+      switch (*argv[1]) {
```
подключите нормальную библиотеку для работы с аргументами командной строки

Hamsterrhino replied:
Порекомендуйте "нормальную" библиотеку тогда? Просто не уверен какую взять? (На примете только boost/program_options)

drewxa commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,111 @@
+        if (sqlite3_exec(db, sql, 0, 0, &err)) {
+          std::cout << stderr << "������ SQL: " << err << std::endl;
+
+          sqlite3_free(err);
```
наверное тут ошибка и не надо выполнять код после if ?

Hamsterrhino replied:
Правда ваша, исправил

drewxa commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,111 @@
+        sql = _strdup("CREATE TABLE IF NOT EXISTS Passwords(SOURCE TEXT NOT "
+                      "NULL, LOGIN TEXT,PASSWORD TEXT );");
+        if (sqlite3_exec(db, sql, 0, 0, &err)) {
+          std::cout << stderr << "������ SQL: " << err << std::endl;
```
кодировка поехала

Hamsterrhino replied:
Исправил

drewxa commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,111 @@
+        const char *login = argv[3];
+        std::string password;
+        std::cin >> password;
+        unsigned char *encryptedpass =
```
std::vector

Hamsterrhino replied:
Тоже самое что и с key, я из алгоритма получаю unsigned char*, переводить это в вектор уж очень не удобно.

drewxa commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,111 @@
+        cout << encryptedpass << endl;
+        std::string sqstring =
+            "INSERT INTO Passwords "
+            "('Source','Login','Password') VALUES (@0,@1,@2);";
```
пробелы после ,

Hamsterrhino replied:
Исправил, вроде.

drewxa commented on Password Manager/Password Manager.cpp:
```diff
@@ -0,0 +1,111 @@
+        sqlite3_bind_text(stmt, 3, (const char *)encryptedpass, -1, 0);
+        sqlite3_step(stmt);
+        // Closing DB
+        sqlite3_close(db);
```
Я все же либо за:
 * использовать готовую обертку над sqlite3, чтобы не париться с низкоуровневым интерфейсом
 * или написать свою обертку под свои нужды
 * или хотя бы использовать unique_ptr чтобы он автоматом освобождал ресурсы за вами

Hamsterrhino committed:
Fixed few requests

Hamsterrhino committed:
Quality-of-reading improvement

Added comments to most of the action
Moved AES and sqlite related files to Third-party
Broke main code onto multiple functions.
